### PR TITLE
Componentise explorer all

### PR DESCRIPTION
**Pull request by NakulManchanda**
- creating this pull request to compare changes and as placeholder for final componentization work

### CODE REVIEW

File: scripts/run-zowe.template.sh [hunk 1]
```diff
@@ -111,7 +107,7 @@ then
   mkdir -p ${STATIC_DEF_CONFIG_DIR}
 
   # Until ui explorers componentised will copy them from the old location
-  cp ${ROOT_DIR}/components/api-mediation/api-defs/* ${STATIC_DEF_CONFIG_DIR}
+  # cp ${ROOT_DIR}/components/api-mediation/api-defs/* ${STATIC_DEF_CONFIG_DIR}
```
Comment (stevenhorsman) [inline scripts/run-zowe.template.sh]: Please delete this line and the comment above

File: scripts/utils/validateNode.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# This program and the accompanying materials are made available under the terms of the
+# Eclipse Public License v2.0 which accompanies this distribution, and is available at
+# https://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Copyright IBM Corporation 2019
+################################################################################
+
+# NODE_HOME Should contain a valid install of Node
```
Comment (stevenhorsman) [inline scripts/utils/validateNode.sh]: Consider checking for the correct version in this script

File: scripts/zowe-install-explorer-ui.sh [hunk 3]
```diff
@@ -18,27 +18,58 @@
...
-  mkdir -p "${ZOWE_ROOT_DIR}/${EXPLORER_INSTALL_FOLDER}"
+  mkdir -p "${EXPLORER_INSTALL_FOLDER}/bin"
+  
   # unpax package
-  cd "${ZOWE_ROOT_DIR}/${EXPLORER_INSTALL_FOLDER}"
+  cd "${EXPLORER_INSTALL_FOLDER}/bin"
   echo "  Unpax of ${EXPLORER_PLUGIN_PAX} into ${PWD}" >> $LOG_FILE
   pax -rf $EXPLORER_PLUGIN_PAX -ppx
+
+  #TODO make sure scripts end up in files directory not bin.
```
Comment (stevenhorsman) [inline scripts/zowe-install-explorer-ui.sh]: What is this TODO for?

File: scripts/zowe-install-explorer-ui.sh [hunk 4]
```diff
@@ -18,27 +18,58 @@
...
+    mv ${EXPLORER_UI_CONFIGURE_SCRIPT} configure.sh
+  fi
+
+  rm -rf $EXPLORER_INSTALL_FOLDER/bin/scripts
+
+
+  # if [[ -f ${EXPLORER_UI_VALIDATE_SCRIPT} ]]
+  # then
+  #   cp ${EXPLORER_UI_VALIDATE_SCRIPT} validate.sh
+  # fi
```
Comment (stevenhorsman) [inline scripts/zowe-install-explorer-ui.sh]: Any reason these are commented out?

File: scripts/configure/zowe-configure-iframe-plugins.sh [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    # Access API Catalog with token injector
+    CATALOG_GATEWAY_URL=https://$ZOWE_EXPLORER_HOST:$ZOWE_ZLUX_SERVER_HTTPS_PORT/ZLUX/plugins/org.zowe.zlux.auth.apiml/services/tokenInjector/1.0.0/ui/v1/apicatalog/
+else
+    # Access API Catalog directly
+    CATALOG_GATEWAY_URL=https://$ZOWE_EXPLORER_HOST:$ZOWE_APIM_GATEWAY_PORT/ui/v1/apicatalog
+fi
+
+# Add API Catalog application to zLUX - required before we issue ZLUX deploy.sh
+# TODO - move into apiml config? run before deploy?
+. $CONFIG_DIR/zowe-install-iframe-plugin.sh $ZOWE_ROOT_DIR "org.zowe.api.catalog" "API Catalog" $CATALOG_GATEWAY_URL $ZOWE_ROOT_DIR"/components/api-mediation/api-catalog.png"
```
Comment (stevenhorsman) [inline scripts/configure/zowe-configure-iframe-plugins.sh]: Consider move the api-catalog iframe plugin into the api-medation component for consistency

File: scripts/zowe-install-explorer-ui.sh (hunk 4 above)
Comment (NakulManchanda) [reply scripts/zowe-install-explorer-ui.sh]: placeholder for validate script, adding validation script so un-comment these

File: scripts/run-zowe.template.sh (hunk 1 above)
Comment (NakulManchanda) [reply scripts/run-zowe.template.sh]: removed

File: scripts/utils/validateNode.sh (hunk 2 above)
Comment (NakulManchanda) [reply scripts/utils/validateNode.sh]: add minimum version check of 6.00, not checking

File: scripts/configure/zowe-configure-iframe-plugins.sh (hunk 5 above)
Comment (NakulManchanda) [reply scripts/configure/zowe-configure-iframe-plugins.sh]: we can move this, may be separate PR, after explorer componentization

File: scripts/zowe-install-explorer-ui.sh (hunk 3 above)
Comment (NakulManchanda) [reply scripts/zowe-install-explorer-ui.sh]: i guess comment remained there from initial planning as scripts do end up in bin folder now

File: scripts/configure/zowe-configure-iframe-plugins.sh [hunk 6]
```diff
@@ -0,0 +1,78 @@
+
```
Comment (stevenhorsman) [inline scripts/configure/zowe-configure-iframe-plugins.sh]: Add shebang

File: scripts/utils/validateNode.sh [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        . ${ROOT_DIR}/scripts/utils/error.sh "NODE_HOME: ${NODE_HOME}/bin does not point to a valid install of Node";
+    else
+
+      NODE_OK=`${NODE_HOME}/bin/node -e "console.log('ok')" 2>&1`
+      if [[ ! $NODE_OK == "ok" ]];
+      then 
+        . ${ROOT_DIR}/scripts/utils/error.sh "${NODE_HOME}/bin/node is not functioning correctly";
+      fi
+
+      NODE_MIN_VERSION=6.0
```
Comment (stevenhorsman) [inline scripts/utils/validateNode.sh]: The minimum version of node is 6.14.4.1 not 6.0

File: scripts/utils/validateNode.sh [hunk 8]
```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
```
Comment (stevenhorsman) [inline scripts/utils/validateNode.sh]: Rename this file to validate-node.sh

File: scripts/utils/validateNode.sh (hunk 7 above)
Comment (NakulManchanda) [reply scripts/utils/validateNode.sh]: bumped it to 6.14, adding check till 4th place

File: manifest.json.template [hunk 9]
```diff
@@ -53,13 +53,13 @@
       "explode": "true"
     },
     "org.zowe.explorer-jes": {
```
Comment (stevenhorsman) [inline manifest.json.template]: Please back out these changes before merging

Comment (NakulManchanda) [reply manifest.json.template]: changed to jes , uss , mvs latest staging version

**Pull request merged**